File: flutter_essentials/lib/app_info.dart
import 'package:flutter_essentials/flutter_essentials.dart';

class AppInfo {
  static const String _methodPrefix = "AppInfo.";

  static Future<String> get packageName async {
    final String version = await SHARED_CHANNEL.invokeMethod('${_methodPrefix}packageName');
    return version;
  }

  static Future<String> get name async {
    final String version = await SHARED_CHANNEL.invokeMethod('${_methodPrefix}name');
    return version;
  }

  static Future<String> get versionString async {
    final String version = await SHARED_CHANNEL.invokeMethod('${_methodPrefix}versionString');
    return version;
  }

  static Future<String> get buildString async {
    final String version = await SHARED_CHANNEL.invokeMethod('${_methodPrefix}buildString');
    return version;
  }

  static Future get showSettingsUI async {
    final String version = await SHARED_CHANNEL.invokeMethod('${_methodPrefix}showSettingsUI');
    return version;
  }
}

File: ios/Classes/AppInfo.swift
import Foundation
import Flutter

public class AppInfo : NSObject{
    static var prefix = "AppInfo."
    
    static func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        if let range = call.method.range(of: AppInfo.prefix){
            let method = call.method[range.upperBound...]
            if method == "packageName"{
                result(getBundleValue(key: "CFBundleIdentifier"))
            }else if method == "name"{
                let name: String = getBundleValue(key: "CFBundleDisplayName") ?? getBundleValue(key: "CFBundleName")!
                result(name)
            }
            else if method == "versionString"{
                result(getBundleValue(key: "CFBundleShortVersionString"))
            }else if method == "buildString"{
                result(getBundleValue(key: "CFBundleVersion"))
            }
            else if method == "showSettingsUI"{
                platformShowSettingsUI()
                result("")
            }
            else{
                result(FlutterMethodNotImplemented)
            }
        }else{
            result(FlutterMethodNotImplemented)
        }
    }
    
    static func getBundleValue(key : String) -> String?{
        return Bundle.main.object(forInfoDictionaryKey: key) as? String;
    }
    
    static func platformShowSettingsUI(){
        guard let settingsUrl = URL(string: UIApplicationOpenSettingsURLString) else {
            return
        }
        
        UIApplication.shared.openURL(settingsUrl)
    }
}
